### PR TITLE
Task/add lakectl local pull tests 6380

### LINKED ISSUE #6380: lakectl local: Esti tests for pull

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #6380

### CODE REVIEW

File: esti/lakectl_local_test.go [hunk 1]
```diff
@@ -169,11 +186,99 @@ func TestLakectlLocal_clone(t *testing.T) {
...
+			RunCmdAndVerifyContainsText(t, Lakectl()+" local pull "+dataDir, false, successStr, vars)
+			verifyDirContents(t, dataDir, expected)
+
+			// Modify data
+			vars["FILE_PATH"] = modified[0]
+			runCmd(t, Lakectl()+" fs upload -s files/ro_1k_other lakefs://"+vars["REPO"]+"/"+vars["BRANCH"]+"/"+vars["FILE_PATH"], false, false, vars)
+			runCmd(t, Lakectl()+" fs rm lakefs://"+repoName+"/"+vars["BRANCH"]+"/"+deleted[0], false, false, vars)
+			runCmd(t, Lakectl()+" commit lakefs://"+vars["REPO"]+"/"+vars["BRANCH"]+" --allow-empty-message -m \" \"", false, false, vars)
+			expected = slices.DeleteFunc(expected, func(s string) bool {
+				return slices.Contains(deleted, s)
```
Comment (nopcoder) [inline esti/lakectl_local_test.go:274]: The deleted is one item - it means we don't need a slice of `deleted` as the code also run a single command and reference `deleted[0]`.
So having when we delete the item from the slice we can just:
```suggestion
				return s == deleted
```

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: esti/lakectl_local_test.go [hunk 2]
```diff
@@ -169,11 +186,99 @@ func TestLakectlLocal_clone(t *testing.T) {
...
+		{
+			name:   "prefix",
+			prefix: "images",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dataDir, err := os.MkdirTemp(tmpDir, "")
+			require.NoError(t, err)
+			vars["PREFIX"] = "/" + tt.prefix
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply esti/lakectl_local_test.go:235]: There's some setup code that's running before the tests and requires the values. I don't feel this is necessary

File: cmd/lakectl/cmd/local_checkout.go [hunk 3]
```diff
@@ -99,7 +99,7 @@ func localCheckout(ctx context.Context, localPath string, syncFlags syncFlags, s
 	}
 
 	summary := syncMgr.Summary()
-	fmt.Printf("Checkout Summary:\nDownloaded:\t%d\nRemoved:\t%d\n", summary.Downloaded, summary.Removed)
+	fmt.Printf("Checkout Summary:\nDownloaded:%d\nRemoved:%d\n", summary.Downloaded, summary.Removed)
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply cmd/lakectl/cmd/local_checkout.go]: I'll add colors in a different PR

File: esti/lakectl_local_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (nopcoder) [reply esti/lakectl_local_test.go:236]: Except repo, storage and branch the setup code doesn't provide extra information for the vars.

**Review by nopcoder — APPROVED** [review]
LGTM!

**Pull request merged**